modules/core/src/test/java/org/apache/ignite/util/GridLogThrottleTest.java

```diff
@@ -34,82 +38,127 @@ public GridLogThrottleTest() {
         LT.warn(log, "Test - without throwable1.");
+        assertTrue(log.toString().contains("Test - without throwable1."));
+        assertEquals(4, LT.errorsSize());
+        log.reset();
 
         // OMITTED.
         LT.warn(log, "Test - without throwable1.");
+        assertTrue(log.toString().contains(""));
+        assertEquals(4, LT.errorsSize());
+        log.reset();
 
-        Thread.sleep(LT.throttleTimeout());
+        U.sleep(LT.throttleTimeout() * 2);
+        info("Slept for throttle timeout: " + LT.throttleTimeout() * 2);
 
-        info("Slept for throttle timeout: " + LT.throttleTimeout());
+        // LOGGED.
+        LT.error(log, new RuntimeException("Test exception 1."), "Test msg");
+        assertTrue(log.toString().contains("Test msg" + sep + "java.lang.RuntimeException: Test exception 1."));
+        assertEquals(1, LT.errorsSize());
+        log.reset();
+
+        // OMMITED.
+        LT.error(log, new RuntimeException("Test exception 1."), "Test msg");
+        assertEquals(log.toString(), "");
+        assertEquals(1, LT.errorsSize());
+        log.reset();
 
         // LOGGED.
-        LT.error(log, new RuntimeException("Test exception 1."), "Test");
+        LT.info(log, "Test info message.");
+        assertTrue(log.toString().contains("Test info message."));
+        assertEquals(2, LT.errorsSize());
+        log.reset();
 
         // OMITTED.
-        LT.error(log, new RuntimeException("Test exception 1."), "Test");
+        LT.info(log, "Test info message.");
+        assertTrue(log.toString().contains(""));
+        assertEquals(2, LT.errorsSize());
+        log.reset();
 
-        // OMITTED.
-        LT.error(log, new RuntimeException("Test exception 1."), "Test1");
+        U.sleep(LT.throttleTimeout() * 2);
+        info("Slept for throttle timeout: " + LT.throttleTimeout() * 2);
 
-        // LOGGED.
-        LT.error(log, new RuntimeException("Test exception 2."), "Test");
+        assertEquals(0, LT.errorsSize());
+    }
 
-        // LOGGED.
-        LT.warn(log, "Test - without throwable.");
+    /**
+     * Test check changing throttleTimeout.
+     *
+     * @throws Exception If any error occurs.
+     */
+    public void testOnChangingTimeout() throws Exception {
+        final GridStringLogger log = new GridStringLogger(false, this.log);
 
-        // OMITTED.
-        LT.warn(log, "Test - without throwable.");
+        LT.throttleTimeout(200);
 
-        Thread.sleep(LT.throttleTimeout());
+        LT.error(log, new RuntimeException("Test exception 1."), "Test");
+        assertEquals(1, LT.errorsSize());
 
-        info("Slept for throttle timeout: " + LT.throttleTimeout());
+        LT.error(log, new RuntimeException("Test exception 1."), "Test");
+        assertEquals(1, LT.errorsSize());
 
-        //LOGGED.
-        LT.info(log(), "Test info message.");
+        U.sleep(LT.throttleTimeout() * 2);
+        info("Slept for throttle timeout: " + LT.throttleTimeout() * 2);
```

Comment:
Can we change info message?
The info is about that we slept for timeout, but we slept twice as much.

Reply:
Done.